test/conformance/ingress/util.go

```diff
@@ -844,12 +845,16 @@ func CreateDialContext(t *testing.T, ing *v1alpha1.Ingress, clients *test.Client
 }
 
 type RequestOption func(*http.Request)
+type Expectations struct {
```

Comment:
I'd rather see expectation to be
```
...  expectations []func(http.Response)error...

...

for _, e := range expectations {
  if e(resp) != nil {
    // fail test
  }
}
...
```

Reply:
Perhaps instead of error we should make the function(s) "accept" the condition?

e.g. some "accept" a status code (200, 404), and we'd want one that "accepts" dial errors

Returning an error and failing the first one makes this harder IMO because to get the effect we want you can't just pass a few built-in functions, e.g.

`RuntimeRequestWithStatus(t, client, url, AllowStatus(404), AllowDialError)`